src/main/kotlin/com/valensas/ftp/model/FTPSClient.kt

```diff
@@ -0,0 +1,49 @@
+package com.valensas.ftp.model
+
+import org.apache.commons.net.ftp.FTPSClient
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.io.IOException
+import java.net.SocketException
+
+class FTPSClient(
+    private val isImplicit: Boolean = false,
+) : FTPClient() {
+    private val logger: Logger = LoggerFactory.getLogger(javaClass)
+
+    private val ftpsClient = FTPSClient(isImplicit)
+
+    @Throws(IOException::class, SocketException::class)
+    override fun authAndConnect(connectionModel: ConnectionModel) {
+        try {
+            ftpsClient.connect(connectionModel.host, connectionModel.port)
+            ftpsClient.login(connectionModel.username, connectionModel.password)
+        } catch (e: Throwable) {
+            logger.error("An error occured while connecting to FTPS server", e)
+            throw e
+        }
+    }
+
+    @Throws(IOException::class)
+    override fun disconnect() {
```

Comment:
disconnect ve sendCommandın neden ayrıca fonksiyonunu yazdık

Reply:
Çünkü ftpsClient'ın ayrı disconnect ve sendCommand metodları var:

```kotlin
public void disconnect() throws IOException {
        super.disconnect();
        if (this.plainSocket != null) {
            this.plainSocket.close();
        }

        this.setSocketFactory((SocketFactory)null);
        this.setServerSocketFactory((ServerSocketFactory)null);
    }

public int sendCommand(String command, String args) throws IOException {
      int repCode = super.sendCommand(command, args);
      if ("CCC".equals(command)) {
          if (200 != repCode) {
              throw new SSLException(this.getReplyString());
          }

          this._socket_.close();
          this._socket_ = this.plainSocket;
          this._controlInput_ = new BufferedReader(new InputStreamReader(this._socket_.getInputStream(), this.getControlEncoding()));
          this._controlOutput_ = new BufferedWriter(new OutputStreamWriter(this._socket_.getOutputStream(), this.getControlEncoding()));
      }

      return repCode;
  }
  
  ```
 Yukarıdaki metodlar Apache kütüphanesindeki FTPSClient'tın metodları. `super.disconnect()` ve `super.sendCommand()` ile parent class'ın metodlarını çağırdıktan sonra FTPS için yaptığı bazı spesifik şeyler var. @KaanSimsek19

Reply:
Burada benim oluşturduğum FTPSClient, Apache'nin FTPSClient'dan türemiyor. Benim yazdığım FTPClient'tan türüyor.

```kotlin
class FTPSClient(
    private val isImplicit: Boolean = false,
) : FTPClient() {
    private val logger: Logger = LoggerFactory.getLogger(javaClass)

    private val ftpsClient = FTPSClient(isImplicit)

```

FTPClient da Apache nin FTPClient'ınından türüyor

```kotlin
import org.apache.commons.net.ftp.FTPClient

open class FTPClient : FTPClient() {
```

Benim oluşturduğum `FTPSClient`'ın Apache'nin `FTPSClient`'ının methodlarını kullanabilmesi için bir `ftpsClinet = FTPSClient(isImplicit)` objesi oluşturuyorum. Eğer bunu eklemezsem ve bu metodları override etmezsem, benim oluşturduğum `FTPSClient` class'ım, normal `FTPClient` gibi davranır @KaanSimsek19

Reply:
Doğru haklısın :)